lib/vdc/DCBOV.cpp

```diff
@@ -0,0 +1,329 @@
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <iostream>
+#include <fstream>
+#include "vapor/VAssert.h"
+#include <stdio.h>
+
+#ifdef _WINDOWS
+    #define _USE_MATH_DEFINES
+    #pragma warning(disable : 4251 4100)
+#endif
+#include <cmath>
+
+#include <vapor/GeoUtil.h>
+#include <vapor/UDUnitsClass.h>
+#include <vapor/DCBOV.h>
+#include <vapor/DCUtils.h>
+
+using namespace VAPoR;
+using namespace std;
+
+DCBOV::DCBOV() : _bovCollection(nullptr)
+{
+    _dimsMap.clear();
+    _coordVarsMap.clear();
+    _dataVarsMap.clear();
+    _meshMap.clear();
+    _coordVarKeys.clear();
+}
+
+DCBOV::~DCBOV()
+{
+    if (_bovCollection != nullptr) { delete _bovCollection; }
+}
+
+int DCBOV::initialize(const vector<string> &paths, const std::vector<string> &options)
+{
+    _bovCollection = new BOVCollection();
+    int rc = _bovCollection->Initialize(paths);
+    if (rc < 0) {
+        SetErrMsg("Failure reading .bov file");
+        return (-1);
```

Comment:
Throughout this code, error is sometime detected by `if (rc == ERROR)`, while other time it's detected in the fashion above. I suggest we strive for consistency within the BOV module.

Reply:
I know what you mean.  The enum is only for parsing though.  Not sure what to do here.

Reply:
Think about it from a readability perspective. It's easy to under stand if you have all integers:
```c++
int ret = func1();
if( ret != 0 ) { // do something}
```

It's also easy to understand if you have all enum values:
```c++
CODE ret = func2();
if( ret != CODE::GOOD ) { // do something }
```
So a reviewer can know that one's comparing ints to ints, or enums to enums. However, it's quite confusing to mix them up:
```c++
int ret = func3();
if( ret != CODE::GOOD ) { // do something }
```

Reply:
Hmm.  I agree with you but I think this might be a bigger discussion.  

This is the same thing that's done everywhere else in the application.  If we had an enum class to define GOOD and BAD, wouldn't we want it up higher in the VAPoR namespace, not here?  I think we'd want to make a new rule for it in our coding conventions too.  Is it alright if I file a ticket for this that we can discuss later?

Reply:
I think you're half way there to achieve a consistent style in the BOV module, so you should finish it. The motivation is that the code currently brings confusion when you compare an integer to an enum. I always need to go look at the enum definition and do a translation between the two values you're comparing in my mind. 

As far as what VAPOR should be doing outside of the scope of the BOV reader, we can certainly discuss later.

Reply:
OK, but one question: where would you define the enum class?  I haven't investigated this yet.  DCBOV and BOVCollection both need to make use of it.  Should they both have their own identical copies of the enum class?

If we're doing down this route now, we need to consider that every return code in Vapor should be refactored to use it for consistency.  I feel like this is getting outside of the scope of this PR, but if you can make a suggestion I can follow through.

I need to go AFK for a while right now but if you can make a suggestion I can get to it when I return.

Reply:
I think it should live in the VAPoR namespace so anyone who wants to use it can use it. BOV module is definitely one of the customers. 

Should we expand it to other customers? Probably should. Should it be now? Maybe not. 

This is something I have that you can use as a starting point. Of course you can add/remove items depending on your needs.
```c++
enum class RTNType { // Return Type
    Good,          
    WrongSize,
    IOError,
    InvalidParam,
    BitBudgetMet,
    VersionMismatch,
    ZSTDMismatch,
    ZSTDError,
    DimensionMismatch,
    Error
};
```

Reply:
Fixed in BOVCollection.